src/wayland/text_input/text_input_handle.rs

```diff
@@ -77,6 +81,18 @@ impl TextInputHandle {
         self.inner.lock().unwrap().focus = surface;
     }
 
+    fn set_enabled_resource_id(&self, resource_id: Option<ObjectId>) {
+        let mut inner = self.inner.lock().unwrap();
+
+        match (&inner.enabled_resource_id, &resource_id) {
+            (Some(_), Some(_)) => {}
+            (None, None) => {}
+            (_, _) => {
+                inner.enabled_resource_id = resource_id;
+            }
+        }
```

Comment:
Maybe that is just me, but I find this easier to read:

```suggestion
        if inner.enabled_resource_id.is_some() != resource_id.is_some() {
            inner.enabled_resource_id = resource_id;
        }
```